### PR TITLE
Body block allowed with weapons

### LINKED ISSUE #46671: Krav Maga Block Counter irregularities
# Describe the bug
Krav Maga **Block Counter** does not proc with style weapons like the USP and combat knife.

# Steps To Reproduce
1. Create a player with an USP, combat knife, and tiger claws
2. Learn Krav Maga
3. Set Unarmed skill to 4+
4. Find something that hits back
3. Try to block using above weapons

# Expected behavior
Player occasionally blocks incoming damage using the USP or combat knife, if I'm reading [this](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/techniques.json#L960) correctly.

# Screenshots
Log shows *other* Krav Maga techniques being used, but no Block Counters proc
kravmagalog

# Versions and configuration
 - OS: Windows 10
 - Game Version: 0.E-8852-gf360c7f
 - Graphics version: Tiles
 - Ingame language: English
 - Mods loaded: dda, no_npc_food

# Additional context
Tiger Claws and other unarmed seem to proc arm- as well as leg blocks as expected.

### PR DESCRIPTION
**Pull request by Termineitor244**
#### Summary
Content "Adds a new flag for weapons to allow blocking with your body parts even while wielding them"

#### Purpose of change
Many weapons are small enough for allowing blocking with your body while using them, but this is not currently allowed in the game, where you can only block with your body while unarmed, or block with your weapon.
Fixes #46671 (Already closed for staled)

#### Describe the solution
Adds a new flag that allows you to block (if your martial art allows it) with your body even while wielding a specific weapon, this flag was added to shivs, knives, pistols and other small items that should not impede you from using your arms/legs to block an attack.

#### Describe alternatives you've considered
None.

#### Testing
Compiled the game, tested using Krav Maga to block with your body while wielding a knife... It works!

#### Additional context
imagen
imagen

### CODE REVIEW

**Review by Maleclypse — COMMENTED** [review]
Needs the flag explained in one of the markdown docs. But really cool!

**Comment by Termineitor244** [comment]
> Needs the flag explained in one of the markdown docs. But really cool!

I had completely forgotten about the docs, thanks for the reminder! I will add an explanation to the docs when I return from work today

**Comment by Dan1ss1mo** [comment]
Do you need help with docs by any chance? Would love to see this pr merged before next stable

**Comment by Termineitor244** [comment]
> Do you need help with docs by any chance? Would love to see this pr merged before next stable

Hi! Sorry, I was quite busy after my last comment in here so I didn't have time to even keep up with the latest additions to the main game, I already added the change to the docs, but I knew very well while doing this PR that it would not be merged until after stable, first and foremost because this is a c++ change to add a new functionality to weapons/martial arts, while we were (and still are) under string freeze (includes content and feature freezes), and second because, well, this PR has new strings to translate in the form of the new flag hahaha.

That said, I would be glad to see it included in the next stable if possible.

**Comment by Dan1ss1mo** [comment]
> > Do you need help with docs by any chance? Would love to see this pr merged before next stable
> 
> Hi! Sorry, I was quite busy after my last comment in here so I didn't have time to even keep up with the latest additions to the main game, I already added the change to the docs, but I knew very well while doing this PR that it would not be merged until after stable, first and foremost because this is a c++ change to add a new functionality to weapons/martial arts, while we were (and still are) under string freeze (includes content and feature freezes), and second because, well, this PR has new strings to translate in the form of the new flag hahaha.
> 
> That said, I would be glad to see it included in the next stable if possible.

ah, thats a shame, a great job nonetheless!

File: data/json/items/resources/misc.json [hunk 1]
```diff
@@ -250,8 +250,8 @@
     "material": [ "flesh" ],
     "volume": "250 ml",
     "cutting": 8,
-    "flags": [ "SPEAR", "SHEATH_KNIFE" ],
-    "weapon_category": [ "SHIVS" ],
+    "flags": [ "SPEAR", "SHEATH_KNIFE", "ALLOWS_BODY_BLOCK" ],
+    "weapon_category": [ "SHIVS", "ALLOWS_BODY_BLOCK" ],
```
Comment (Dan1ss1mo) [inline data/json/items/resources/misc.json:254]: "weapon_category" probably shouldn't have "ALLOWS_BODY_BLOCK" flag)

**Comment by Dan1ss1mo** [comment]
for some weird reason, it seems that you become unable to block at all with MA's that have "prevent_weapon_blocking": true line

**Comment by Dan1ss1mo** [comment]
Yet if i remove it, i seem to only block with my limbs with "force_unarmed": true ma's, whatever I'm unarmed, with normla weapon or with ALLOWS_BODY_BLOCK weapon

**Comment by Dan1ss1mo** [comment]
Huh, apperently it isn't coused by this pr, it seems that this part cause it 
```else if( !allow_weapon_blocking ) {
        // Can't block with weapons
        return false;```

File: data/json/items/resources/misc.json (hunk 1 above)
Comment (Termineitor244) [reply data/json/items/resources/misc.json:254]: Thank you! I had not realized I typed the flag where it didn't belong...

**Comment by NetSysFire** [comment]
>(Already closed for staled)

I just took care of that.

**Comment by Termineitor244** [comment]
> for some weird reason, it seems that you become unable to block at all with MA's that have "prevent_weapon_blocking": true line, even then unarmed



> Huh, apperently it isn't coused by this pr

Yeah, I just tested in my local copy of the game and the bug is there, so at least it doesn't has anything to do with my PR, but this is another bug that should be resolved before stable... Probably in another PR, I will call it to attention in the discord...

And why did you close the issue you made about this bug?... #63271

**Comment by Termineitor244** [comment]
> > (Already closed for staled)
> 
> I just took care of that.

Thanks!

**Comment by Dan1ss1mo** [comment]
> > for some weird reason, it seems that you become unable to block at all with MA's that have "prevent_weapon_blocking": true line, even then unarmed
> 
> > Huh, apperently it isn't coused by this pr
> 
> Yeah, I just tested in my local copy of the game and the bug is there, so at least it doesn't has anything to do with my PR, but this is another bug that should be resolved before stable... Probably in another PR, I will call it to attention in the discord...
> 
> And why did you close the issue you made about this bug?... #63271

Ah, I got a bit confused about whatever or not i was able to reproduce it on latest vanilla version, should definitely reopen it)

**Comment by Dan1ss1mo** [comment]
> > for some weird reason, it seems that you become unable to block at all with MA's that have "prevent_weapon_blocking": true line, even then unarmed
> 
> > Huh, apperently it isn't coused by this pr
> 
> Yeah, I just tested in my local copy of the game and the bug is there, so at least it doesn't has anything to do with my PR, but this is another bug that should be resolved before stable... Probably in another PR, I will call it to attention in the discord...
> 
> And why did you close the issue you made about this bug?... #63271

Seems like I was able to fix it. Also tested the fix with you're pr, everything seems to work as intended

**Pull request merged**